### ISSUE TITLE
Crashing on A12X device(s).

### ISSUE THREAD

**Issue by fnschmidt** [CoryKornowicz/Alticha-voucher_swap-Razmashat-RootKit-voucher_swap#3]
Using iPad8,1 (11" WiFi 256) on 12.1.1. After tapping "gain root here" it always hangs for 5-10 secs and then reboots. This was tried about fifteen times in total including the new "Badges" which should also work on A12 but doesn't seem to on A12X.

**Comment by CoryKornowicz** [comment]
I am using the Alticha voucher_swap here, I know that it can be shaky with anything that isn't A12, I will update with the supposed new rootKit that came out. But from my understanding, there is not an adaptation of the exploit for A12X devices anywhere currently. That might change with the rootless JB coming out soon, but for now, none of the voucher_swaps are currently working with A12X.

**Comment by fnschmidt** [comment]
Do you know if this is an issue because the A12X is substantially enough different (security wise?) or if this is just an issue atm because of low interest?

**Comment by CoryKornowicz** [comment]
I think it's just for the moment. Just as A7/A8 wasn't in the first port of voucher_swap but there are devs getting it to work now. It is kinda spread about right now throughout where everyone is working on something new. I have a feeling it won't be too much trouble to add. It might be an offset issue, I'm not sure for certain.

**Comment by fnschmidt** [comment]
I see, thank you for the response. I was only curious why so many people are talking about A7/8 and pretty much none about A12X but thinking about it, iPhone 6 was one of the best selling iPhones and iPads don‘t get bought nearly as much. Well, I hope voucher_swap gets updated and a root rw remount exploit published soon but as always no one is entitled to anything.

**Comment by CoryKornowicz** [comment]
Me too, I think that we will see voucher_swap updated to support all devices before we get that sweet full jailbreak just because people are anxious and they probably want to try out the badge switcher and things like that. But, the ball is rolling and each day we're getting closer.

**Comment by fnschmidt** [comment]
Yeah, I just read about the CoreTrust bypass by IBSparkes that will probably be released soon. If things keep moving at this pace we might see a JB for a signed iOS version!

**Comment by CoryKornowicz** [comment]
Yea this is legendary pace of a jailbreak development, just make sure to get those blobs hahaha

**Comment by fnschmidt** [comment]
Yeah I have every blob saved except 12.1 because my first iPad had a broken power button and the second one shipping with 12.1.1, but I can‘t imagine a 12.1 only JB so I don‘t really care. On a sidenote I loathe Apple so much for so perfectly preventing downgrades, I even regularly still buy old 32 bit devices to install legacy iOS versions on them for nostalgia‘s sake.

**Comment by CoryKornowicz** [comment]
I still have my iPhone 4s around with 6.1.5 I think, the good ol’ days

**Comment by fnschmidt** [comment]
What I theoretically even could see Apple doing was locking downgrades to $99 dev accounts and adding some changes to make the version “uncomfortable“ for most users, like disabling Touch/FaceID, making the boot screen bright red and disabling some iCloud stuff or whatever, and then justificate that by only having it for development purposes and not to compromise your Apple ID security or whatever. I just hate that I will never be able to reexperience some old iOS version as I‘m a huge nerd for that.

**Comment by CoryKornowicz** [comment]
I could totally see that, but if they managed to even open up downgrades to the dev accounts...we’d bypass it somehow. They’re safe right now because no loophole exists except for their internal tools and such, but I mean they opened signing applications to developers on their own and we got Cydia Impactor. Even look at futureRestore. A small crack like that and we’d blow it open.

**Comment by fnschmidt** [comment]
Yeah and I also think Apple really has no incentive to make this happen, as Apple doesn‘t exactly want app developers to develop for old iOS versions. (At least I assume, I mean even a >1 year old iOS version has incompatibility issues, whereas on the Android side you are pretty much above average if your OS is only 1 year old).

Also if we are already offtopic, future iOS update support could become really interesting: up to now old devices were killed when they were quite slow, but that kind of will have to change: the 6s on 12 is just as old as the 4 on iOS 7, but the 4 was horrible even with many disabled features like blur and parallax while the 6s is pretty much excellent on 12, and no way in hell will a mobile OS in the foreseeable future will take up all resources on say a A12 or even the beastly A12X. I‘m sure the current SoC have so much spare power they could support new iOS for a whole decade, so at some point Apple will start killing off devices that could run the new OS but are just plain too old.

**Comment by CoryKornowicz** [comment]
I mean besides their planned obsolescence and purposeful underclocking older devices to reduce battery ware or whatever excuse they have, I could totally see an iOS coming up where the A12 chip is challeneged. Think about the most aggressive jailbreak tweaks and put them into the OS. Let’s say splitview multitasking, manually enabling which apps run for forever in the background, a constant live running wallpaper. I heard somewhere that Apple plans about 3 years ahead so I could only imagine the behemoths of board designs they have. There was a YouTube documentary on Apple that talked about them testing out their own processors in Macs instead of intel and that means cross compatibility to iPads at the least. Imagine MacOS on an iPad. That’s why I think that there will always come a time when and iPhone becomes weak, it’s just when will it happen.

**Comment by fnschmidt** [comment]
Well, I see that iOS devices get slower. It just appeared to me that the duration for which each device still was fast has been steadily increasing. I mean a 5s is on its 6th OS, and newer devices will (or could) last much longer than 6 years for a single reason: the OS requirement increase is only slowly creeping up over the years (iOS 12 doesn‘t use that much more than 9 etc.) but each CPU generation introduced a much higher power increase (excluding A7>A8 and maybe A11>A12 on the CPU side). So the exponential performance growth is IMO outrunning the slower iOS requirement growth. So even a major upgrade like (hopefully) iOS 13 probably won‘t have that much of an impact on newer CPUs if it still works good on an A9 for example.
I think Apple will introduce an ARM based MacBook in 2020/21 because it would make sense- the iPads currently have more power and a far lower power consumption. Also Apple has the control over the software development chain to make it happen, which is what Microsoft can’t really do.
But I don‘t think Apple will introduce macOS to the iPad because Apple wouldn‘t „allow“ users to use such a complex UI on an iPad, and also wouldn‘t introduce something like Samsung DeX to have a full macOS using mouse and KB simply because a) both of these options aren‘t „apple-like“ (too complicated etc.) and b) it would cannibalize Mac sales and the iPad would have to become much more pricey.
